demo/api/src/importer/importers/product-importer/product-importer.ts

```diff
@@ -0,0 +1,54 @@
+import { EntityManager } from "@mikro-orm/core";
+import { Injectable, Logger, LoggerService } from "@nestjs/common";
+import { DataStream, DataStreamAndMetaData } from "@src/importer/data-streams/data-stream";
+import { pipeline, Transform } from "stream";
+
+@Injectable()
+export class ProductImporter {
+    dataStream: DataStreamAndMetaData | null = null;
+    name = "rawProductImport";
+    transformPipes: Transform[] = [];
+    logger: LoggerService;
+
+    constructor(private readonly em: EntityManager) {
+        this.logger = new Logger("product-importer");
```

Comment:
The logger should be initialized outside of the constructor per the Nest docs:

```ts
@Injectable()
class MyService {
  private readonly logger = new Logger(MyService.name);
```

https://docs.nestjs.com/techniques/logger#using-the-logger-for-application-logging